sdk/python/feast/feature_store.py

```diff
@@ -153,6 +167,76 @@ def get_historical_features(
         )
         return job
 
+    def materialize(
+        self,
+        feature_views: Optional[List[str]],
+        start_date: datetime,
+        end_date: datetime,
+    ) -> None:
+        """
+        Materialize data from the offline store into the online store.
+
+        This method loads feature data in the specified interval from either
+        the specified feature views, or all feature views if none are specified,
+        into the online store where it is available for online serving.
+
+        Args:
+            feature_views (List[str]): Optional list of feature view names. If selected, will only run
+                materialization for the specified feature views.
+            start_date (datetime): Start date of query
+            end_date (datetime): End date of query
+        """
+        feature_views_to_materialize = []
+        registry = self._get_registry()
+        if feature_views is None:
+            feature_views_to_materialize = registry.list_feature_views(
+                self.config.project
+            )
+        else:
+            for name in feature_views:
+                feature_view = registry.get_feature_view(name, self.config.project)
+                feature_views_to_materialize.append(feature_view)
+
+        # TODO paging large loads
+        for feature_view in feature_views_to_materialize:
+            if isinstance(feature_view.input, FileSource):
+                raise NotImplementedError(
+                    "This function is not yet implemented for File data sources"
+                )
+            if not feature_view.input.table_ref:
+                raise NotImplementedError(
+                    f"This function is only implemented for FeatureViews with a table_ref; {feature_view.name} does not have one."
+                )
+            (
+                entity_names,
+                feature_names,
+                event_timestamp_column,
+                created_timestamp_column,
+            ) = _run_reverse_field_mapping(feature_view)
+
+            offline_store = self._get_offline_store()
+            table = offline_store.pull_latest_from_table(
+                feature_view.input,
+                entity_names,
+                feature_names,
+                event_timestamp_column,
```

Comment:
Is there still a reason to have the columns as part of this method? (`event_timestamp_column`, `created_timestamp_column`)

Reply:
yes, these are the post-feature-mapping names, and I want the feature-mapping to happen outside the BigQuery-specific method which means these names have to be passed in